[Replicate] Simplify code a bit

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #98890
* __->__ #98889

Simplifies the code, such as making self.modules not a list and only a
single module.

Differential Revision: [D44899281](https://our.internmc.facebook.com/intern/diff/D44899281/)

rohan-varma committed:
[Replicate] Simplify code a bit

Simplifies the code, such as making self.modules not a list and only a
single module.

Differential Revision: [D44899281](https://our.internmc.facebook.com/intern/diff/D44899281/)

[ghstack-poisoned]

mrshenli commented on torch/distributed/_composable/replicate.py:
```diff
@@ -34,23 +34,21 @@ def _can_compose(module: nn.Module) -> bool:
 class _ReplicateState:
     def __init__(self) -> None:
-        self.modules: List[nn.Module] = []
+        self.module = None
```
nit: keep `nn.Module` typing?

mrshenli reviewed (approved):
LGTM. Please fix lint before landing :)

yhcharles reviewed (approved):
Thanks, LGTM with a minor comment

rohan-varma committed:
Update on "[Replicate] Simplify code a bit"


Simplifies the code, such as making self.modules not a list and only a
single module.

Differential Revision: [D44899281](https://our.internmc.facebook.com/intern/diff/D44899281/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Replicate] Simplify code a bit"


Simplifies the code, such as making self.modules not a list and only a
single module.

Differential Revision: [D44899281](https://our.internmc.facebook.com/intern/diff/D44899281/)

[ghstack-poisoned]

rohan-varma commented:
@pytorchbot merge -f "CI done"
